packages/lesswrong/server/vulcan-lib/apollo-server/authentication.tsx

```diff
@@ -330,6 +331,23 @@ function registerLoginEvent(user, req) {
     currentUser: user,
     validate: false,
   })
+  
+  const clientId = getCookieFromReq(req, "clientId");
+  if (clientId) {
+    void recordAssociationBetweenUserAndClientID(clientId, user);
+  }
+}
+
+async function recordAssociationBetweenUserAndClientID(clientId: string, user: DbUser) {
+  const clientIdEntry = await ClientIds.findOne({clientId});
```

Comment:
The chance of a race condition actually occurring here is minuscule, but for what it's worth I think the correct way to do this is:
```
ClientIds.findAndModify({clientId}, {$addToSet: { userIds:  userId }})
```
probably some small performance bonus for doing this as well since there's only one database round-trip.

Reply:
jk I just tried this and it's hard to do with Vulcan. Nevermind.